timemachine/datasets/utils.py

```diff
@@ -3,24 +3,29 @@
 
 from rdkit import Chem
 
-from timemachine.fe.utils import get_mol_name
+from timemachine.fe.utils import get_mol_name, read_sdf
 
 
 def fetch_freesolv(n_mols: Optional[int] = None, exclude_mols: Optional[Set[str]] = None) -> List[Chem.Mol]:
     """
-    Return the (potentially truncated) free solv data set.
+    Return the (potentially truncated) FreeSolv data set.
 
     Parameters
     ----------
     n_mols:
-        Limit to this number of mols. Default of None
-        means to keep all of the molecules.
+        Limit to this number of mols.
+        Default of None means to keep all of the molecules.
 
     exclude_mols:
         Exclude molecules in the given set.
 
     """
-    exclude_mols = exclude_mols or set()
     with resources.path("timemachine.datasets.freesolv", "freesolv.sdf") as freesolv_path:
-        supplier = Chem.SDMolSupplier(str(freesolv_path), removeHs=False)
-    return [mol for mol in supplier if get_mol_name(mol) not in exclude_mols][:n_mols]
+        mols = read_sdf(str(freesolv_path))
+
+    # filter and truncate
+    exclude_mols = exclude_mols or set()
+    filtered_mols = [mol for mol in mols if get_mol_name(mol) not in exclude_mols]
+    first_n_filtered_mols = filtered_mols[:n_mols]
+
+    return first_n_filtered_mols
```

Comment:
Just noting that I think this interface can mostly be replicated with one-liner list comprehensions now that we have `read_sdf`, e.g.

```python
mols = list(islice(mol for mol in read_sdf("ligands.sdf") if get_mol_name(mol) not in {"mol_1", "mol_2"}))
```

Reply:
I agree -- I was just reproducing and splitting up the the current behavior of `fetch_fressolv`